feat: add reveal guide (#71)
Basis-Theory/developers.basistheory.com: docs/guides/share/reveal-tokenized-data.mdx

    }

    // this is just an example method that'll have the full lifecycle for revealing
    private fun reveal() {
      // highlight-next-line
      val session = bt.createSession()
    }
}
```
  </TabItem>
</Tabs>

## Authorizing a Session

In order to use the session to retrieve data, we need to authorize it with a Private Application.

### Creating a Private Application

First, lets create the [Private Application](docs/api/applications#application-types). To do so, [Login to your Basis Theory account](https://portal.basistheory.com/applications) and create a new application with the following settings:

- Name - Reveal Data
- Application Type - Private
- Access Rule (Use the `Advanced Rule Builder` option)
  - Description - Reveal Data
  - Container - `/`
  - Permissions - `token:read`
  - Transform - `reveal`

<Alert>
    The Private Application must be in the same <a href="/docs/concepts/access-controls#what-are-tenants">Tenant</a> as the token to be revealed.
</Alert>

<Alert>
  Save the API Key from the created Private Application as it will be used later in this guide.
</Alert>

### Authorizing In the Backend

> Now we can use the created Private Application API Key, and the `nonce` from our created session to authorize it.
> We'll also use a [Condition](/docs/concepts/access-controls#conditions) to make sure our session is only authorized to reveal our desired token.

Reviewer: ```suggestion
Using the Private Application API Key, the `nonce` from our session, and [Condition](/docs/concepts/access-controls#conditions) we ensure we are only granting authorization for the desired token.
```

> This step <b>SHOULD</b> be done on the backend as to not expose the Private Application's API Key publicly.

Reviewer: ```suggestion
The *only way* to ensure your private API keys are not publically accessible is for this step to execute in your backend service.
```

We'll create a `backend.js` file and add the following code to start the Express.js backend and authorize an incoming session.

```javascript showLineNumbers title="backend.js"
const express = require("express");
const { BasisTheory } = require("@basis-theory/basis-theory-js");

const app = express();
const port = 4242;

app.post("/authorize", async (request, response) => {
  const bt = await new BasisTheory().init("test_priv_1234567890");
  const { nonce } = request.body;
  // authorizing a session returns an empty 200 response
  await bt.sessions.authorize({
    nonce: nonce,
    rules: [
      {
        description: "Reveal Token",
        priority: 1,
        conditions: [
          {
            attribute: "id",
            operator: "equals",
            value: "token_id_1234567890",
          },
        ],
        permissions: ["token:read", "token:use"],
        transform: "reveal",
      },
    ],
  });
  // this response is arbitrary and not required
  response.json({
    result: "success",
  });
});
app.listen(port, () => {
  console.log(`Example app listening on port ${port}`);
});
```

> Now we can start the server with the following command (from the same directory as `backend.js`).

Reviewer: ```suggestion
Start the server with the following command (from the same directory as `backend.js`).
```

        exampleTextElement = findViewById(R.id.example_text_element)
    }

    // this is just an example method that'll have the full lifecycle for revealing
    private fun reveal() {
      val session = bt.createSession()

      authorizeSession(session.nonce)

      //highlight-next-line
      val token = bt.getToken("token_id_1234567890", session.sessionKey)

      //highlight-next-line
      exampleTextElement.setValueRef(token.data)
    }

    private fun authorizeSession(nonce: String) {
      val url = URL("http://localhost:4242/authorize")
      val con = url.openConnection() as HttpURLConnection

      con.requestMethod = "POST"
      con.doOutput = true
      con.setRequestProperty("Content-Type", "application/json")
      con.setRequestProperty("Accept", "application/json");

      val body = String.format("{\"nonce\": \"%s\"}", nonce)

      con.outputStream.use { os ->
          val input: ByteArray = body.toByteArray(Charsets.UTF_8)
          os.write(input, 0, input.size)
      }

      if (con.responseCode == 200) {
          return
      }
  }
}
```
  </TabItem>
</Tabs>

🎉 The code above is the last bit that we need to reveal a token! 

## Conclusion

> You can now reveal any data to a customer without your iOS app accessing the underlying value, drastically reducing compliance and regulatory scope.

Reviewer: ```suggestion
You can now reveal any data to a customer without your applications accessing the underlying value, reducing compliance and regulatory scope.
```

Author: oopsie

      <TextElement
        id="exampleTextElement"
        ref={textRef}
      />
    </BasisTheoryProvider>
  );
}

export default App;
```
  </TabItem>
  <TabItem value="ios" label="iOS">

```swift showLineNumbers
import UIKit
import BasisTheoryElements

class ViewController: UIViewController {
    @IBOutlet weak var exampleTextElement: TextElementUITextField!

    // this is just an example method that'll have the full lifecycle for revealing
    func reveal() {
      BasisTheoryElements.createSession(apiKey: btPublicKey) { data, error in
            let sessionKey = data!.sessionKey!
            let nonce = data!.nonce!

            
            self.authorizeSession(nonce: nonce) { result, error in
              //highlight-start
              BasisTheoryElements.getTokenById(id: 'token_id_1234567890', apiKey: sessionKey) { data, error in
                  self.exampleTextElement.setValue(elementValueReference: data!.data!.elementValueReference)
              }
              //highlight-end
            }
      }
    }

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
    }

>     //highlight-start

Reviewer 2: we want to remove this highlight right? since we've already highlighted this in the step before?
